utils/merge_and_push_translations.sh

```diff
@@ -79,64 +147,36 @@ if [ -n "$(git status -s)" ]; then
   # Push source catalogs to Transifex
   tx push -s
   sleep 65 # wait for translation files to be updated
-fi
 
-# Pull All translations from Transifex
-# tx pull -a
-# Force-pull translations because some would get skipped accidentally
-tx pull -fa
-
-# Update authors file
-python3 utils/update_authors.py
-if [ $? -eq 0 ] ; then
-  echo "Updated authors";
-else
-  echo "Failed updating authors";
-  exit 1;
-fi
+  # Pull All translations from Transifex
+  # use force to make sure really all files get pulled
+  tx pull -a -f
 
-# Update appdata
-python3 utils/update_appdata.py
-if [ $? -eq 0 ] ; then
-  echo "Updated appdata";
-else
-  echo "Failed updating appdata";
-  exit 1;
+  # Undo one-liner diffs of pure timestamps with no other content
+  undo_oneliner_diffs
+  runUpdateScripts=2 # run, files have changed
 fi
 
-# Update statistics
-python3 utils/update_translation_stats.py
-if [ $? -eq 0 ] ; then
-  echo "Updated translation stats";
-else
-  echo "Failed to update translation stats";
-  exit 1;
+if [ -n "$runUpdateScripts" ]; then # if not run yet or pot file has changed
+  update_authors # in case something has changed unexpectedly
+  update_appdata # in case something has changed unexpectedly
+  update_statistics
 fi
 
-# Fix formatting for Lua
-python3 utils/fix_formatting.py --lua --dir data/i18n
-python3 utils/fix_formatting.py --lua --dir data/txts
-
-# Undo one-liner diffs of pure timestamps with no other content
-undo_oneliner_diffs
-
 if [ -z "$(git status -s)" ]; then
   echo "Run completed, nothing to commit."
-  exit 0
+else
+  # Stage changes
+  # - Translations and templates
+  git add 'po/*/*.po' 'po/*/*.pot' 'data/i18n/locales/*.json' 'xdg/translations/*.json' || true
+  # - generated files
+  gitAddGeneratedFiles
+
+  # Commit
+  git commit -m "Updated translations catalogs and statistics."
 fi
 
-# Stage changes
-# - Translations
-git add po/*/*.po po/*/*.pot data/i18n/locales/*.json xdg/translations/*.json || true
-# - Authors
-git add data/txts/*.lua || true
-# - Locale data
-git add data/i18n/*.lua || true
-# - Appdata
-git add xdg/org.widelands.Widelands.appdata.xml xdg/org.widelands.Widelands.desktop || true
-# - Statistics
-git add data/i18n/translation_stats.conf || true
-
-# Commit and push.
-git commit -m "Fetched translations and updated catalogs."
-git push "$push_target" master
+if [ "$(git show --no-patch --format=format:_ FETCH_HEAD HEAD)" != "_" ]; then # check if it is the same commit
```

Comment:
You could set `HAD_COMMIT=yes` on commits to make this test simpler.

Reply:
Shouldn't I trust git? :wink: I will adapt, makes it more obvious.

Reply:
I know. :) I meant it as "if you like", for readability. But it's OK this way too.

Reply:
I did :+1: for readability.

Reply:
Now we have to trust the programmer to set has_commit=yes after a commit was done.
We could save the the hash of HEAD in the beginning and check in the end if it is still the same. This should be more readable than my previous git command

Reply:
> But there should not be many new commits created, and the programmers are trustworthy.

This is only inside this script. Not too hard to verify correctness.